NDTensors/ext/NDTensorsMetalExt/set_types.jl

```diff
@@ -39,9 +39,26 @@ default_parameter(::Type{<:MtlArray}, ::Position{3}) = Metal.DefaultStorageMode
 
 nparameters(::Type{<:MtlArray}) = Val(3)
 
+using NDTensors.TypeParameterAccessors: TypeParameterAccessors
 # Metal-specific type parameter setting
-function set_storagemode(arraytype::Type{<:MtlArray}, storagemode)
-  return set_parameter(arraytype, Position(3), storagemode)
+function set_storagemode(arraytype::Type{<:MtlArray}, param)
+  return TypeParameterAccessors.set_type_parameter(arraytype, NDTensors.storagemode, param)
 end
 
 SetParameters.unspecify_parameters(::Type{<:MtlArray}) = MtlArray
+
+function TypeParameterAccessors.position(::Type{<:MtlArray}, ::typeof(eltype))
+  return TypeParameterAccessors.Position(1)
+end
+function TypeParameterAccessors.position(::Type{<:MtlArray}, ::typeof(Base.ndims))
+  return TypeParameterAccessors.Position(2)
+end
+function TypeParameterAccessors.position(
+  ::Type{<:MtlArray}, ::typeof(NDTensors.storagemode)
```

Comment:
```suggestion
  ::Type{<:MtlArray}, ::typeof(storagemode)
```
and same thing elsewhere.

Reply:
This is updated across the board thanks!